### PR TITLE
Fix deprecation warning from docker_py

### PR DESCRIPTION
**Pull request by anish**
Squashes : "Passing host config parameters in start() is deprecated. Please use host_config in create_container instead."

### CODE REVIEW

**Comment by anish** [comment]
This has been deprecated as of docker-py 1.2.3, current version is 1.7.2

File: appveyor.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ environment:
 install:
   # Twisted requires pywin32 on Windows in order to spawn processes.
   - "%PYTHON%\\python.exe -m pip install pypiwin32"
+  - "%PYTHON%\\python.exe -m pip install docker-py"
```
Comment (rutsky) [inline appveyor.yml]: Don't forget to remove this line, since Fake Docker client should be used.

**Comment by rutsky** [comment]
@anish, please add to message [here](https://github.com/anish/buildbot/blob/add_host_config/master/buildbot/worker/docker.py#L67) that `docker-py>=1.4.0` is required.

File: master/buildbot/test/fake/docker.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def __init__(self, base_url):
     def images(self):
         return self._images
 
-    def start(self, *args, **kwargs):
+    def start(self, instance):
```
Comment (rutsky) [inline master/buildbot/test/fake/docker.py]: I think argument should be called `container`: https://docker-py.readthedocs.org/en/latest/api/#start

**Comment by rutsky** [comment]
Other than `instance` -> `container` renaming this PR LGTM, thanks!

**Comment by anish** [comment]
@rutsky but the code that starts this calls it instance : 
```
        self.instance = instance
        docker_client.start(instance)
        log.msg('Container started')
```

Either is okay with me, but please confirm

**Comment by rutsky** [comment]
@anish, more important is that `container` is part of `start()` function definition in `docker-py`, and we should mimic it in our Fake implementation.

This is an example of `start()` usage from `docker-py` documentation:

```
>>> from docker import Client
>>> cli = Client(base_url='tcp://127.0.0.1:2375')
>>> container = cli.create_container(
...     image='busybox:latest',
...     command='/bin/sleep 30')
>>> response = cli.start(container=container.get('Id'))
>>> print(response)
None
```
and it explicitly uses `container` as keyword argument, which will not work in your implementation.

**Comment by anish** [comment]
gotcha. Changed as requested.

**Comment by rutsky** [comment]
LGTM!

**Pull request merged**